content/post/2021-07-29-caching/index.Rmd

```diff
@@ -0,0 +1,187 @@
+---
+slug: cache
+title: "How to cache results in R code" 
+authors: 
+- Maëlle Salmon 
+- Christophe Dervieux
+date: "2021-07-29" 
+tags: 
+- package development 
+output: hugodown::hugo_document
+---
+
+```{r setup, include=FALSE}
+
+knitr::opts_chunk$set(fig.path = "", comment = "")
+# knitr hook to make images output use Hugo options
+knitr::knit_hooks$set(
+  plot = function(x, options) {
+    hugoopts <- options$hugoopts
+    paste0(
+      "{{<figure src=",
+      '"', x, '" ',
+      if (!is.null(hugoopts)) {
+        glue::glue_collapse(
+          glue::glue('{names(hugoopts)}="{hugoopts}"'),
+          sep = " "
+        )
+      },
+      ">}}\n"
+    )
+  }
+)
+
+```
+
+One principle of programming that's often encountered is "DRY", "Don't Repeat Yourself", that encourages e.g. the use of functions other duplicated (read: copy-pasted and slightly amended) code.
+You could also interpret it as don't let the machine repeat its calculations if useless.
+How about for a piece of code (function) with the same inputs, we only run it once per R session, and save the results for later?
+In this post, we shall go over ways to cache results in an R session, so that you don't need to burden machines.
+We will not cover [caching for R Markdown](https://bookdown.org/yihui/rmarkdown-cookbook/cache.html).
```

Comment:
Or maybe it could be a short section with pointers to the relevant recipes of the cookbook for instance (here I only link to one recipe but I see there are three relevant recipes).

Reply:
Good point, I have now reduced the scope and mentioned these elements in the conclusion.